lib/fog/ovirt/requests/compute/v4/vm_start_with_cloudinit.rb

```diff
@@ -0,0 +1,23 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          def vm_start_with_cloudinit(options = {})
+            raise ArgumentError, "instance id is a required parameter" unless options.key? :id
+
+            vm_service = client.system_service.vms_service.vm_service(options[:id])
+            vm_service.start(:use_cloud_init => true, :vm => options[:user_data])
```

Comment:
While testing I've noticed this was not actually working. After checking with docs, the right line should be:

```ruby
vm_service.start(:use_cloud_init => true, :vm => { :initialization => options[:user_data] })
```

What I was using for checking with Foreman devel instance (given I had some shutdown VM with specific UUID):

```
ci = <<EOF
hostname: test.example.com
ssh_authorized_keys:
- ssh-rsa test123123123 test.example.com
runcmd:
- ls /
phone_home:
  url: satellite.exxample.com
  post: []
EOF
cr = ComputeResource.find_by(name: 'ovirt')
vm = cr.find_vm_by_uuid('1d1c0451-a686-403b-91ae-57f300f80c59')
vm.start_with_cloudinit(:blocking => true, :user_data => ci, :use_custom_script => true)
```

And after the vm booted:

```
mkdir /media/ci
mount /media/ci /dev/sr1
```

And checking that the /media/ci had the user data I've passed there at the beginning.